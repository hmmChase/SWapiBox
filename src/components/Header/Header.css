.Header {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  justify-content: space-between;
  align-items: center;
  padding: 0 50px 0 50px;
  border-bottom: 2px solid yellow;
  background-image: url(https://thumbs.gfycat.com/IgnorantDimpledHyrax-size_restricted.gif);
  background-repeat: no-repeat;
  background-size: cover;
}

.title {
  color: #000;
  font-family: 'Audiowide', sans-serif;
  -webkit-text-stroke: 1px yellow;
}

.categories {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin: 0 2em 0 2em;
}

.num-favs {
  margin-left: 5px;
}

.navlink {
  padding: 10px;
  border: 2px solid yellow;
  border-radius: 10px;
  margin: 0.5em;
  background: none;
  color: #fff;
  font-size: 1.5em;
  font-weight: 600;
  text-decoration: none;
  cursor: pointer;
}

.navlink:hover {
  background: rgba(164, 164, 0, 0.5);
}

.active {
  color: #feda4a;
}
