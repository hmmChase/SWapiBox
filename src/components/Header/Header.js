import React from 'react';
import { NavLink } from 'react-router-dom';
import './Header.css';

const Header = ({ numFavs }) => {
  return (
    <section className="Header">
      <h1>SWapiBox</h1>
      <div className="categories">
        <NavLink className="navlink" to="/people">
          People
        </NavLink>
        <NavLink className="navlink" to="/planets">
          Planets
        </NavLink>
        <NavLink className="navlink" to="/vehicles">
          Vehicles
        </NavLink>
      </div>

      <NavLink className="navlink" to="/favorites">
        Favorites
        <span>:{numFavs}</span>
      </NavLink>
    </section>
  );
};

export default Header;
