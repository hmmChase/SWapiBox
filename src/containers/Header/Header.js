import React from 'react';
import './Header.css';

const Header = () => {
  return <section className="Header">Header</section>;
};

export default Header;
