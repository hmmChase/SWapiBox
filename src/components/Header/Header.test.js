import React from 'react';
import { shallow, mount } from 'enzyme';
import Header from './Header';

describe('Header', () => {
  let header;
  beforeEach(() => {
    header = shallow(<Header />);
  });

  it('matches snapshot', () => {
    expect(header).toMatchSnapshot();
  });
});
