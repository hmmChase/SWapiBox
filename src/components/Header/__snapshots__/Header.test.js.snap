// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    numFavs={2}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          SWapiBox
</h1>,
        <div
          className="categories"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="navlink"
                    to="/people"
          >
                    People
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="navlink"
                    to="/planets"
          >
                    Planets
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="navlink"
                    to="/vehicles"
          >
                    Vehicles
          </NavLink>
</div>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="navlink"
          to="/favorites"
>
          Favorites
          <span>
                    :
                    2
          </span>
</NavLink>,
      ],
      "className": "Header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SWapiBox",
        },
        "ref": null,
        "rendered": "SWapiBox",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="navlink"
              to="/people"
>
              People
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="navlink"
              to="/planets"
>
              Planets
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="navlink"
              to="/vehicles"
>
              Vehicles
</NavLink>,
          ],
          "className": "categories",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "People",
              "className": "navlink",
              "to": "/people",
            },
            "ref": null,
            "rendered": "People",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Planets",
              "className": "navlink",
              "to": "/planets",
            },
            "ref": null,
            "rendered": "Planets",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Vehicles",
              "className": "navlink",
              "to": "/vehicles",
            },
            "ref": null,
            "rendered": "Vehicles",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": Array [
            "Favorites",
            <span>
              :
              2
</span>,
          ],
          "className": "navlink",
          "to": "/favorites",
        },
        "ref": null,
        "rendered": Array [
          "Favorites",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                ":",
                2,
              ],
            },
            "ref": null,
            "rendered": Array [
              ":",
              2,
            ],
            "type": "span",
          },
        ],
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            SWapiBox
</h1>,
          <div
            className="categories"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="navlink"
                        to="/people"
            >
                        People
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="navlink"
                        to="/planets"
            >
                        Planets
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="navlink"
                        to="/vehicles"
            >
                        Vehicles
            </NavLink>
</div>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="navlink"
            to="/favorites"
>
            Favorites
            <span>
                        :
                        2
            </span>
</NavLink>,
        ],
        "className": "Header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SWapiBox",
          },
          "ref": null,
          "rendered": "SWapiBox",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="navlink"
                to="/people"
>
                People
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="navlink"
                to="/planets"
>
                Planets
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="navlink"
                to="/vehicles"
>
                Vehicles
</NavLink>,
            ],
            "className": "categories",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "People",
                "className": "navlink",
                "to": "/people",
              },
              "ref": null,
              "rendered": "People",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Planets",
                "className": "navlink",
                "to": "/planets",
              },
              "ref": null,
              "rendered": "Planets",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Vehicles",
                "className": "navlink",
                "to": "/vehicles",
              },
              "ref": null,
              "rendered": "Vehicles",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": Array [
              "Favorites",
              <span>
                :
                2
</span>,
            ],
            "className": "navlink",
            "to": "/favorites",
          },
          "ref": null,
          "rendered": Array [
            "Favorites",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  ":",
                  2,
                ],
              },
              "ref": null,
              "rendered": Array [
                ":",
                2,
              ],
              "type": "span",
            },
          ],
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
